tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -1293,6 +1298,292 @@ def test_supported_bucket_policy_operations(self, mcg_obj, bucket_factory):
             not missing_policies
         ), f"Some bucket_policies are not created : {missing_policies}"
 
+    @staticmethod
+    def check_commands(
+        mcg_obj,
+        awscli_pod_session,
+        bucket_name,
+        path_on_bucket,
+        file_name,
+        allow_ls_anonymous,
+    ):
+        """
+        Check that 'ls' and 'cp' commands on the bucket work as expected.
+        The expected behavior of 'ls' is:
+            1. Success and finding 'file_name' file with non-anonymous access
+            2. Success with anonymous access when 'allow_anonymous' is True
+            3. Failure with "Access Denied" on anonymous access when 'allow_anonymous' is False
+        The expected behavior of 'cp' is:
+            1. Success with non-anonymous access
+            2. Failure with "Access Denied" on anonymous access
+        Args:
+            mcg_obj (obj): An object representing the current state of the MCG in the cluster
+            awscli_pod_session (pod): A pod running the AWSCLI tools
+            path_on_bucket (str) Target path on the bucket, to which the file is copied. If empty, the files
+                    is copied to the bucket root.
+            bucket_name (str): Name of the bucket on which ls should be run
+            file_name (str): File to be looked for
+            allow_ls_anonymous (bool): Defines whether 'ls' with anonymous access is allowed
+        Raises:
+            UnexpectedBehaviour if the ls is not working as expected
+        """
+
+        ls_command = f"ls s3://{bucket_name}/{path_on_bucket}"
+        cp_command = f"cp {AWSCLI_TEST_OBJ_DIR}{file_name} s3://{bucket_name}/{path_on_bucket}{file_name}"
+
+        awscli_pod_session.exec_cmd_on_pod(
+            command=craft_s3_command(cp_command, mcg_obj=mcg_obj),
+            out_yaml_format=False,
+        )
+
+        ls_output = awscli_pod_session.exec_cmd_on_pod(
+            command=craft_s3_command(ls_command, mcg_obj=mcg_obj),
+            out_yaml_format=False,
+        )
+        if file_name not in ls_output:
+            raise UnexpectedBehaviour(
+                f"ls command doesn't show {file_name} as expected"
+            )
+
+        if allow_ls_anonymous:
+            awscli_pod_session.exec_cmd_on_pod(
+                command=craft_s3_command(
+                    f"{ls_command} --no-sign-request", mcg_obj=mcg_obj
+                ),
+                out_yaml_format=False,
+            )
+        else:
+            try:
+                awscli_pod_session.exec_cmd_on_pod(
+                    command=craft_s3_command(
+                        f"{ls_command} --no-sign-request", mcg_obj=mcg_obj
+                    ),
+                    out_yaml_format=False,
+                )
```

Comment:
why is the same command tried twice in if else block?

Reply:
@keesturam Rewrote the code , now this command appears only once.